Mention StatesPlugin in the migration docs (#1378)
bevyengine/bevy-website: release-content/0.14/migration-guides/13216_Separate_state_crate.md

Projects that use `bevy_ecs` directly and use states will need to add the `bevy_state` crate as a dependency.

Projects that use `bevy_app` directly and use states will need to add the `bevy_state` feature.

> To enable state functionality to your app, you will need to add the `StatesPlugin`.

Reviewer: ```suggestion
If you do not use `DefaultPlugins`, you will need to add the `StatesPlugin` manually to your app.
```
